datalad/support/annexrepo.py

```diff
@@ -829,7 +829,8 @@ def is_special_annex_remote(self, remote, check_if_known=True):
             if remote not in self.get_remotes():
                 raise RemoteNotAvailableError(remote)
         opts = self.config.options('remote.{}'.format(remote))
-        if "url" in opts:
+        # for LFS git-annex records URL in the config
+        if "url" in opts and ('annex-git-lfs' not in opts):
```

Comment:
We've discussed in the past that `is_special_annex_remote` needs work to deal with remotes that can be both regular git remotes and special remotes (gh-3499).  I don't think the change here is going in the right direction.  The call in `publish` to `is_special_annex_remote` is used to decide to return early for special remotes so that it does not carry out the fetch/push that it would for regular git remotes.  But lfs remotes are also regular git remotes, so we don't want that logic in `publish` to abort early because `is_special_annex_remote` returned True (as it would after this change).  I think the alternative that you propose in gh-4200 is a better approach.

Reply:
interesting!  in my example case they were two separate remotes `github-lfs` and `github`... I did not dare to name them the same way... and upon clone, initremote I also get two:
```
$> git annex enableremote github-lfs  
enableremote github-lfs ok
(recording state in git...)

$> cat .git/config 
[core]
	repositoryformatversion = 0
	filemode = true
	bare = false
	logallrefupdates = true
[remote "origin"]
	url = https://github.com/yarikoptic/test-github-lfs
	fetch = +refs/heads/*:refs/remotes/origin/*
	annex-ignore = true
[branch "master"]
	remote = origin
	merge = refs/heads/master
[annex]
	uuid = bac78362-ce90-4c9c-980e-d01d5966422a
	version = 5
[remote "github-lfs"]
	annex-git-lfs = true
	annex-uuid = 88a34ed0-2e3a-4103-9bc3-7c4649d71472
	url = https://github.com/yarikoptic/test-github-lfs
```
So, I am not sure if that is the case with LFS special remote that it would be a part of the original one...  and if I force upon clone to use the same name `github-lfs` for the origin, fun happens:
```
$> git annex enableremote github-lfs
enableremote github-lfs 
  Remote github-lfs not usable by git-annex; setting annex-ignore
failed
git-annex: enableremote: 1 failed
```
so github-lfs doesn't get `annex-git-lfs` in its remote config.

So, in this particular case it might still be safe and actually more correct than before because a remote with `annex-git-lfs` is a "special_annex_remote", so this function should return True, not False IMHO.

Reply:
> So, I am not sure if that is the case with LFS special remote that it would be a part of the original one...

git-annex's [LFS documentation][0] has a case where the LFS and the main remote are one and the same:

```
So you can just git clone from the url, and the "origin" remote will
be automatically used as a git-lfs special remote.

    git clone https://github.com/yourname/yourrepo
    cd yourrepo
    git-annex get --from origin

Nice and simple, and much the same as git-annex handles its regular
remotes.
```

[0]: https://git-annex.branchable.com/tips/storing_data_in_git-lfs/

Reply:
ok, some mysteries resolved:

<details>
<summary>such nice way does not work with older 7.20190819+git2-g908476a9b-1~ndall+1 </summary> 

```shell
$> ( rm -rf test-github-lfs && git clone https://github.com/yarikoptic/test-github-lfs && cd test-github-lfs && git annex get 2; git annex version; )
Cloning into 'test-github-lfs'...
remote: Enumerating objects: 51, done.
remote: Counting objects: 100% (51/51), done.
remote: Compressing objects: 100% (37/37), done.
remote: Total 51 (delta 9), reused 49 (delta 7), pack-reused 0
Unpacking objects: 100% (51/51), 4.52 KiB | 771.00 KiB/s, done.
2@  file.dat@  file2.dat@
(merging origin/git-annex into git-annex...)
(recording state in git...)
get 2 
  Remote origin not usable by git-annex; setting annex-ignore
(not available) 
  Try making some of these repositories available:
  	64cdc933-2cca-42ee-9e21-b379d1e83251 -- yoh@lena:/tmp/test-github-lfs
   	88a34ed0-2e3a-4103-9bc3-7c4649d71472 -- github-lfs

  (Note that these git remotes have annex-ignore set: origin)
failed
(recording state in git...)
git-annex: get: 1 failed
git-annex version: 7.20190819+git2-g908476a9b-1~ndall+1

```
</details>

<details>
<summary>but works with newer 7.20200204+git62-gcc4521068-1~ndall+1</summary> 

```shell
$> ( rm -rf test-github-lfs && git clone https://github.com/yarikoptic/test-github-lfs && cd test-github-lfs && git annex get 2; git annex version; )
Cloning into 'test-github-lfs'...
remote: Enumerating objects: 51, done.
remote: Counting objects: 100% (51/51), done.
remote: Compressing objects: 100% (37/37), done.
remote: Total 51 (delta 9), reused 49 (delta 7), pack-reused 0
Unpacking objects: 100% (51/51), 4.52 KiB | 578.00 KiB/s, done.
2@  file.dat@  file2.dat@
(merging origin/git-annex into git-annex...)
(recording state in git...)
(scanning for unlocked files...)
get 2 (from origin...) 
(checksum...) ok
(recording state in git...)
git-annex version: 7.20200204+git62-gcc4521068-1~ndall+1
build flags: Assistant Webapp Pairing S3 WebDAV Inotify DBus DesktopNotify TorrentParser MagicMime Feeds Testsuite
dependency versions: aws-0.20 bloomfilter-2.0.1.0 cryptonite-0.25 DAV-1.3.3 feed-1.0.0.0 ghc-8.4.4 http-client-0.5.13.1 persistent-sqlite-2.8.2 torrent-10000.1.1 uuid-1.3.13 yesod-1.6.0
key/value backends: SHA256E SHA256 SHA512E SHA512 SHA224E SHA224 SHA384E SHA384 SHA3_256E SHA3_256 SHA3_512E SHA3_512 SHA3_224E SHA3_224 SHA3_384E SHA3_384 SKEIN256E SKEIN256 SKEIN512E SKEIN512 BLAKE2B256E BLAKE2B256 BLAKE2B512E BLAKE2B512 BLAKE2B160E BLAKE2B160 BLAKE2B224E BLAKE2B224 BLAKE2B384E BLAKE2B384 BLAKE2BP512E BLAKE2BP512 BLAKE2S256E BLAKE2S256 BLAKE2S160E BLAKE2S160 BLAKE2S224E BLAKE2S224 BLAKE2SP256E BLAKE2SP256 BLAKE2SP224E BLAKE2SP224 SHA1E SHA1 MD5E MD5 WORM URL
remote types: git gcrypt p2p S3 bup directory rsync web bittorrent webdav adb tahoe glacier ddar git-lfs hook external
operating system: linux x86_64
supported repository versions: 7
upgrade supported from repository versions: 0 1 2 3 4 5 6
local repository version: 7

$> cat test-github-lfs/.git/config                                              
[core]
	repositoryformatversion = 0
	filemode = true
	bare = false
	logallrefupdates = true
[remote "origin"]
	url = https://github.com/yarikoptic/test-github-lfs
	fetch = +refs/heads/*:refs/remotes/origin/*
	annex-git-lfs = true
	annex-uuid = 88a34ed0-2e3a-4103-9bc3-7c4649d71472
...
```
</details>
heh heh